tests/test_dynamic_mock.js

```diff
@@ -1,49 +1,51 @@
 'use strict'
 
+const { expect } = require('chai')
 const { test } = require('tap')
-const request = require('request')
 const nock = require('..')
+const got = require('./got_client')
 
 require('./cleanup_after_each')()
+require('./setup')
 
-test('one function returning the status code and body defines a full mock', function(t) {
-  nock('http://example.test')
+test('one function returning the status code and body defines a full mock', async t => {
+  const scope = nock('http://example.test')
     .get('/def')
     .reply(function() {
       return [201, 'DEF']
     })
 
-  request.get('http://example.test/def', function(err, resp, body) {
-    t.error(err)
-    t.equal(resp.statusCode, 201)
-    t.equal(body, 'DEF')
-    t.end()
-  })
+  const { statusCode, body } = await got('http://example.test/def')
+  expect(statusCode).to.equal(201)
+  expect(body).to.equal('DEF')
+
+  scope.done()
+  t.end()
 })
 
-test('one asynchronous function returning the status code and body defines a full mock', function(t) {
-  nock('http://example.test')
+test('one asynchronous function returning the status code and body defines a full mock', async t => {
+  const scope = nock('http://example.test')
     .get('/ghi')
     .reply(function(path, reqBody, cb) {
       setTimeout(function() {
         cb(null, [201, 'GHI'])
       }, 1e3)
     })
 
-  request.get('http://example.test/ghi', function(err, resp, body) {
-    t.error(err)
-    t.equal(resp.statusCode, 201)
-    t.equal(body, 'GHI')
-    t.end()
-  })
+  const { statusCode, body } = await got('http://example.test/ghi')
+  expect(statusCode).to.equal(201)
+  expect(body).to.equal('GHI')
+
+  scope.done()
+  t.end()
 })
 
-test('asynchronous function gets request headers', function(t) {
-  nock('http://example.test')
+test('asynchronous function gets request headers', async t => {
+  const scope = nock('http://example.test')
     .get('/yo')
     .reply(201, function(path, reqBody, cb) {
-      t.equal(this.req.path, '/yo')
-      t.deepEqual(this.req.headers, {
+      expect(this.req.path).to.equal('/yo')
+      expect(this.req.headers).to.include({
```

Comment:
Should this be `to.deepEqual`? Or was this changed because got is injecting more headers?

Reply:
Got adds `user-agent` and `accept-encoding`. Since the user agent includes the version of Got it didn't seem like a good assertion to have.

Reply:
In the Got v10 release notes they said they'd changed this behavior, and are no longer including the version in the user agent: https://github.com/sindresorhus/got/releases/tag/v10.0.0

Reply:
I've updated this, to do a deep equal again, against all the headers.